File: ainpc-core-plugin/src/main/java/ro/ainpc/engine/runtime/ScenarioRuntimeDefinition.java
package ro.ainpc.engine.runtime;

import java.util.LinkedHashMap;
import java.util.Map;

public record ScenarioRuntimeDefinition(
    String id,
    String type,
    Map<String, String> parameters
) {
    public ScenarioRuntimeDefinition {
        id = valueOrEmpty(id);
        type = valueOrEmpty(type);
        parameters = sanitize(parameters);
    }

    public String parameter(String key) {
        return parameters.getOrDefault(key, "");
    }

    private static String valueOrEmpty(String value) {
        return value == null ? "" : value.trim();
    }

    private static Map<String, String> sanitize(Map<String, String> source) {
        if (source == null || source.isEmpty()) {
            return Map.of();
        }
        Map<String, String> sanitized = new LinkedHashMap<>();
        source.forEach((key, value) -> {
            String safeKey = valueOrEmpty(key);
            if (!safeKey.isBlank()) {
                sanitized.put(safeKey, valueOrEmpty(value));
            }
        });
        return Map.copyOf(sanitized);
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/engine/runtime/ScenarioRuntimeDefinition.kt
package ro.ainpc.engine.runtime

import java.util.LinkedHashMap

class ScenarioRuntimeDefinition(
    id: String?,
    type: String?,
    parameters: Map<String, String>?
) {
    private val idValue: String = valueOrEmpty(id)
    private val typeValue: String = valueOrEmpty(type)
    private val parametersValue: Map<String, String> = sanitize(parameters)

    fun id(): String = idValue
    fun type(): String = typeValue
    fun parameters(): Map<String, String> = parametersValue

    fun parameter(key: String?): String = parametersValue[key].orEmpty()

    companion object {
        private fun valueOrEmpty(value: String?): String = value?.trim().orEmpty()

        private fun sanitize(source: Map<String, String>?): Map<String, String> {
            if (source.isNullOrEmpty()) {
                return emptyMap()
            }
            val sanitized = LinkedHashMap<String, String>()
            source.forEach { (key, value) ->
                val safeKey = valueOrEmpty(key)
                if (safeKey.isNotBlank()) {
                    sanitized[safeKey] = valueOrEmpty(value)
                }
            }
            return java.util.Map.copyOf(sanitized)
        }
    }
}
